src/coreclr/vm/threads.cpp

```diff
@@ -7702,6 +7702,23 @@ InterpThreadContext* Thread::GetInterpThreadContext()
 
     return m_pInterpThreadContext;
 }
+
+extern "C" InterpThreadContext* STDCALL GetInterpThreadContextWithPossiblyMissingThread(Thread *pThread)
+{
+    CONTRACTL
+    {
+        THROWS;
+        if (GetThreadNULLOk()) {GC_TRIGGERS;} else {DISABLED(GC_NOTRIGGER);}
```

Comment:
Looking at this again, this looks backwards. I would expect it to be `if (GetThreadNULLOk() == NULL) {GC_TRIGGERS;} else {GC_NOTRIGGER;}`. If the thread exists, this does not trigger GC and just return the thread. If the thread does not exists, this may trigger GC in the process of creating the thread.

However, we have this contract in number of places. I am not sure whether you want to be opening this can of worms in this PR. It may be best to omit this line in this PR.